[py-pybedtools] added version 0.9.0

qwertos opened the pull request.

qwertos committed:
[py-pybedtools] forcing cythanize

qwertos committed:
[py-pybedtools] added version 0.9.0

adamjstewart commented on var/spack/repos/builtin/packages/py-pybedtools/package.py:
```diff
@@ -33,3 +34,6 @@ class PyPybedtools(PythonPackage):
     depends_on("py-six", type=("build", "run"))
+
+    def patch(self):
+        python('setup.py', 'cythonize')
```
Direct invocation of setup.py is [deprecated](https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html). Do the PyPI tarballs come pre-cythonized?

qwertos replied:
no

adamjstewart replied:
@pradyunsg I assume there's no other way to build this package without running `python setup.py cythonize` first?

qwertos replied:
Actually, I might have spoken too soon, let me double check something

qwertos replied:
So, the PyPI tarballs do come pre-cythonized, but the Github tarballs that this package was written with don't. How would you like me to handle this?

adamjstewart replied:
We should always prefer using the PyPI tarballs over the GitHub tarballs: https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#pypi-vs-github

Can you switch from `url` to `pypi` and run `spack checksum` to recompute all the checksums?

qwertos replied:
Done

qwertos committed:
[py-pybedtools]

- Swiched tarball source to pypi
- recomputed all checksums
  - pypi does not host 0.7.3 and 0.7.2. Versions removed.
- Removed patch()

adamjstewart commented on var/spack/repos/builtin/packages/py-pybedtools/package.py:
```diff
@@ -10,19 +10,18 @@ class PyPybedtools(PythonPackage):
+    version('0.7.4',  sha256='15cfae9e8a207ded403ad9fa2e77f09d14c2fe377d1bc5f8b063647e2d0554e0')
+    version('0.6.9',  sha256='56915b3e2200c6fb56260a36f839e66ce27d7dd3ef55fba278c3931b786fbfd1')
 
     depends_on("py-setuptools", type="build")
```
```suggestion
    depends_on("py-setuptools@0.6c5:", type="build")
```

qwertos committed:
[py-pybedtools] set minimum version of setuptools

adamjstewart commented:
@spackbot fix style

qwertos committed:
[@spackbot] updating style on behalf of qwertos
